source/Server/DirectoryServices/DirectoryServicesCredentialValidator.cs

```diff
@@ -94,7 +94,7 @@ internal AuthenticationUserCreateResult GetOrCreateUser(UserValidationResult pri
 
             var users = userStore.GetByIdentity(authenticatingIdentity);
 
-            var existingMatchingUser = users.SingleOrDefault(u => u.Identities != null && u.Identities.Any(identity => 
+            var existingMatchingUser = users.FirstOrDefault(u => u.Identities != null && u.Identities.Any(identity => 
```

Comment:
If this is looking for a "perfect" or "complete" match on all fields, I didn't grok that from the code because of the `FirstOrDefault` and `Any()` calls. They made me think it was looking or any match (partial or full). Perhaps a comment or a code change could help here?